### PR TITLE
Redid DB

### PR DESCRIPTION
**Pull request by RetroSabian**

### CODE REVIEW

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs
Comment (CilliersN) [inline Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs:6]: rather have all you { on the same line as the functions

namespace Know_Your_Nation_Speedy.Models{
     ......
     ......
}

**Review by FredJEhlers — COMMENTED** [review]
I cant see any major problems, Lets see if people disagree

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs
Comment (FredJEhlers) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs:6]: I have experienced some IDEs dont let you choose . Especially Visual Studio

Comment (RetroSabian) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs:6]: will edit, sorry it auto next lines on enter, will keep it cleaner

Comment (FredJEhlers) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs:6]: @RetroSabian I would say leave it since it will be consistent within your project. Dont try fight it, it will constantly fix itself

Comment (FredJEhlers) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs:6]: @CilliersN if you disagree leave the comment if you dont just resolve discussion

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Controllers/UsersController.cs
Comment (RashadAkoodieRabbit) [inline Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Controllers/UsersController.cs:44]: Try to rename "value" to something more appropriate. When you're accessing this method from elsewhere and see that the function takes in "value" it might not make that much sense. Not necessary to change for this PR but good to remember for the future

Comment (RetroSabian) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Controllers/UsersController.cs:44]: Im aware of the controller being poor, only using it for testing purposes, next sprint task is too add and clean up Controllers, thanks for the feedback

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Products.cs
Comment (guandutoit) [inline Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Products.cs]: this should be an icollection

Comment (RetroSabian) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Products.cs]: so the idea is to keep it as an Icollection, but instantiate it as a HashSet?

Comment (RetroSabian) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Products.cs]: would i need to create a new migration if i change this to Icollection?

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs
Comment (CilliersN) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs:6]: nope leave it as is. different IDE's have different standards for c#
no need to make change @RetroSabian

Comment (RashadAkoodieRabbit) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Memberships.cs:6]: I agree with @FredJEhlers on this one

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Nations.cs
Comment (RashadAkoodieRabbit) [inline Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Nations.cs:20]: if you're only initializing within your class, consider making "Article" an internal set. If you're planning to initialize from outside of your class you can leave as is. Same for everywhere else you're initializing the same way

**Review by guandutoit — APPROVED** [review]
Wait for all comments to be resolved before you merge

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Nations.cs
Comment (RetroSabian) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Nations.cs:20]: so what you're saying is declare it without the initialize method? 
eg Icoll Articles = new HashSet();
wont it be better this way, so to save memory for only when the class is used?

Comment (guandutoit) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Nations.cs:20]: what you can do is just initialize it in the constructor instead of calling a method, but you can switch back to the method if you want once you start overloading the constructor

Comment (RetroSabian) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Nations.cs:20]: okay will keep this in mind

**Pull request merged**
